### PR TITLE
allow user to set sample ID in stacks2_ustacks

### PR DESCRIPTION
**Pull request by TomHarrop**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [X] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

---

Hi,

Some Galaxy Australia users want to combine the output of multiple `ustacks` runs at the `cstacks` (catalog building) stage. The current wrapper sets the sample ID for the first sample to 1 and then increments by 1 for each sample. This means sample IDs collide across `ustacks` runs, which causes an error in `cstacks`.

I've added a new parameter (and a test) to `ustacks` to allow the user to set the ID for the first sample. I couldn't think of a lower-impact way of doing it.

Unrelated: one of the existing ustacks tests is failing. I think this might be because the locus ID in the ustacks output changes between runs. I had to modify the tests from diff to `assert_contents`.

Gruß!
Tom

### CODE REVIEW

**Comment by TomHarrop** [comment]
It looks like one of the other tools has issues - I'm adding some more `<assert_contents>`s.

File: tools/stacks2/stacks_ustacks.xml [hunk 1]
```diff
@@ -255,6 +284,23 @@ true
...
         </test>
+        <!-- test setting i -->
+        <test expect_num_outputs="2">
+            <param name="input_type|input_type_select" value="single"/>
+            <param name="input_type|fqinputs" value="demultiplexed/PopA_01.1.fq,demultiplexed/PopA_02.1.fq" ftype="fastqsanger"/>
+            <param name="add_log" value="yes"/>
+            <param name="i" value="42"/>
+            <output name="output_log" ftype="txt"><assert_contents><has_text text="ustacks is done."/></assert_contents></output>
+            <output_collection name="tabs" count="6">
+                <element name="PopA_01.tags"><assert_contents><has_text text="# ustacks completed on "/></assert_contents></element>
```
Comment (bernt-matthias) [inline tools/stacks2/stacks_ustacks.xml:295]: Just a suggestion. You may use a macro for `<element name="PopA_01.tags"><assert_contents><has_text text="# ustacks completed on "/></assert_contents></element>` and parametrize the value of the `name` attribute.

Comment (TomHarrop) [reply tools/stacks2/stacks_ustacks.xml]: Good idea, I have added that macro and applied it where any check was looking for "stacks completed on".

**Comment by bernt-matthias** [comment]
Ready to merge @TomHarrop ?

**Comment by TomHarrop** [comment]
Yes, thanks @bernt-matthias

**Comment by bgruening** [comment]
Thanks @TomHarrop and @bernt-matthias

**Pull request merged**